Stop sending organisations to Rummager, add a SearchPresenter spec

rgarner opened the pull request:
A retrievable-by-author-only restriction in the publishing-api (thanks @tuzz):
https://github.com/alphagov/publishing-api/blob/master/app/controllers/v2/content_items_controller.rb#L64-L68

means that `get_content_items` always returned an empty organisations
list. This means that Rummager never got to relate indexable content to organisation
slugs.

@danielroseman pointed out we could use `get_linkables`
instead, so we stubbed that and added tests for everything else (there was no
 spec for `SearchPresenter`).

Subsequently, @tijmenb pointed out that this will shortly not be necessary,
since 'rummager will start looking up the organisations in the
publishing-api soon-ish'.

As a result, we removed the misleading specs and assertions and left an obvious
placeholder in the `SearchPresenter` spec from which to address this at a time 
when upstream work is complete – but the `get_linkables` commit remains here should
we subsequently need to use it.

Then we removed even that placeholder. We'll wait for Rummager.

rgarner committed:
Make SearchPresenter get real orgs

A retrievable-by-author-only restriction in the publishing-api:
https://github.com/alphagov/publishing-api/blob/master/app/controllers/v2/content_items_controller.rb#L64-L68

means that `get_content_items` always returns an empty organisations
list. This means that Rummager never gets to hear about organisation
slugs and hence can't find our specialist content when including slugs
in search terms.

Daniel Roseman pointed out we should be using `get_linkables`
instead, so we stub that and add tests for everything else (there was no
 spec for SearchPresenter).

Further, Tijmen Brommet pointed out that this will not be necessary,
since 'rummager will start looking up the organisations in the
publishing-api soon-ish'. This is a good thing, as `get_linkables`
for a document type of 'organisation' is very expensive and can't be
filtered.

rgarner committed:
Re-mock every spec that publishes

* SearchPresenter calls get_linkables for orgs now, not get_content
(see previous commit message for details)
* This breaks a *lot* of specs.

rgarner committed:
Remove organisation_slugs from SearchPresenter

Previously, the specs were misleading inasmuch as they were asserting
that we had sent details of organisations to Rummager. Outside of
the stubs in the specs, and in real-world use,
`publishing_api.get_content(document_type: 'organisation'...)` had
begun to return an empty list, so in reality we were never sending
organisation details to Rummager, despite the specs asserting that
we were.

This commit removes the misleading stubs and assertions about what
indexable attributes were sent to Rummager, and leaves a placeholder
for us to address the organisations issue when upstream Rummager work
 around organisation details is complete.

benilovj commented on spec/features/publishing_a_cma_case_spec.rb:
```diff
@@ -86,7 +85,7 @@ def withdrawn_item
 
     publishing_api_has_content([cma_case, minor_update_item, live_item, withdrawn_item], document_type: CmaCase.publishing_api_document_type, fields: fields, page: page_number, per_page: per_page)
-    publishing_api_has_content([cma_org_content_item], document_type: 'organisation', fields: [:base_path, :content_id])
+    publishing_api_has_linkables([cma_org_content_item], document_type: 'organisation')
```
is this still needed?

rgarner replied:
Eek. I missed one. Good spot.

benilovj commented on app/presenters/search_presenter.rb:
```diff
@@ -11,7 +11,11 @@ def to_json
+      # which in turn used publishing_api.get_content to map org content_ids to
+      # organisation slugs. It was stubbed in many places in spec/, but that method
+      # call always returned a blank list in the real world.
+      organisations: [],
```
Since we're working off the assumption that [Rummager will index the organisations from Publishing API](https://trello.com/c/RrN7B6xQ/593-rework-the-way-rummager-indexes-links) very soon, do we need the FIXME and to send the organisations array at all? We can always revert this PR if the assumption turns out to be incorrect for whatever reason.

rgarner replied:
Do we know what the real user impact of not sending this now is? The card you link to seems to suggest the work is a little way off yet.

Happy to take the `organisations: []` line (and the FIXME) out, that was just trying to be explicit (it's meaningless as anything but a placeholder as it gets taken out by the `reject` below).

benilovj replied:
There won't be any user impact, based on the assumption that the Finding Things work will go to production before this tool. If this assumption is no longer true, we will revert this change. I'd prefer to not leave any placeholders in, as that would necessitate another change to clean them out later.

rgarner replied:
I guess I meant "what would the impact be if this were to go live without the Finding Things work?" In theoretical terms I'm curious about what purpose the inflamed appendix we just took out was _for_, but in practical terms I suppose I should just be happy the patient isn't dead. To use an uncomfortably _medical_ metaphor...

tijmenb replied:
Rummager uses the organisations tag to 1) allow filtering by organisation in search and 2) display a metadata thingy in the search result:

<img width="1176" alt="screen shot 2016-05-12 at 12 45 48" src="https://cloud.githubusercontent.com/assets/233676/15213599/2b9ad196-1840-11e6-95f0-18de136896aa.png">

rgarner replied:
Thanks @tijmenb!

benilovj commented on spec/presenters/search_presenter_spec.rb:
```diff
@@ -0,0 +1,45 @@
+      end
+
+      # This needs thought.
+      it 'prepares organisation details *somehow*'
```
Since we're working off the assumption that [Rummager will index the organisations from Publishing API](https://trello.com/c/RrN7B6xQ/593-rework-the-way-rummager-indexes-links) very soon, I don't think we need this.

rgarner replied:
Ah, I misunderstood. Looking at that card, "notifier rather than source of truth" seems to suggest Rummager will just go off and find stuff for itself based on, say, just `content_id` or `base_path`?

benilovj replied:
> "notifier rather than source of truth" seems to suggest Rummager will just go off and find stuff for itself based on, say, just content_id or base_path?

Yes. My understanding is that:
- the publishing tool will publish links to the Publishing API
- Publishing API puts a message on the message queue
- Rummager sees the message on the fanout
- Rummager fetches the links from Publishing API and indexes them

rgarner committed:
Remove the placeholder for orgs in SearchPresenter

* We're asserting that later, Rummager will step in and use the
indexing queue (consumed by rummager-sidekiq) as a notification list
rather than a source of truth, and will be asking the publishing-api
for truth. As a result, we won't need to do anything here (except
perhaps remove further fields to avoid giving the impression that
Rummager will continue to treat anything here as authoritative).
